YIF.Core.Domain/ApiModels/Validators/DirectionPostApiModelValidator.cs

```diff
@@ -0,0 +1,35 @@
+﻿using FluentValidation;
+using System.Linq;
+using System.Resources;
+using YIF.Core.Data;
+using YIF.Core.Domain.ApiModels.RequestApiModels;
+
+namespace YIF.Core.Domain.ApiModels.Validators
+{
+    public class DirectionPostApiModelValidator : AbstractValidator<DirectionPostApiModel>
+    {
+        private readonly EFDbContext _context;
+        private readonly ResourceManager _resourceManager;
+        public DirectionPostApiModelValidator(EFDbContext context, ResourceManager resourceManager)
+        {
+            _resourceManager = resourceManager;
+            _context = context;
+
+            RuleFor(x => x.Name)
+                .NotEmpty()
+                .NotNull();
+
+            RuleFor(x => x.Code)
+                .NotEmpty()
+                .NotNull();
+
+            RuleFor(x => x.Name)
+               .Must(x => _context.Directions.All(n => n.Name != x))
+               .WithMessage(_resourceManager.GetString("AlreadyExistsInDbMessage"));
+
+            RuleFor(x => x.Code)
+                .Must(x => _context.Directions.All(n => n.Code != x))
```

Comment:
why you create validation for code not found in database ? if you creates new direction